### PR TITLE
Fix box to only auto-orient when no direction specified

### PR DESCRIPTION
**Pull request by jieyanhuang**
@yangbin @tgohn @zz85

### CODE REVIEW

File: jx/ui/Box.jxml [hunk 1]
```diff
@@ -9,16 +9,22 @@
...
+					MIN_SIZE = 10,     // minimum element size
+					num_children = 0;  // number of children
+
+				if (direction == 'vertical')
+					flow_dim = 'height',
+					full_dim = 'width',
+					flow_axis = 'y',
+					full_axis = 'x';
 
 				// check if horizontal direction is specified or if width > height to determine direction
```
Comment (yangbin) [inline jx/ui/Box.jxml]: This comment is redundant

File: jx/ui/Box.jxml [hunk 2]
```diff
@@ -9,16 +9,22 @@
...
-				if (direction == 'horizontal' || attr.width > attr.height)
+				if (direction == 'horizontal')
```
Comment (yangbin) [inline jx/ui/Box.jxml]: `else if`

**Comment by jieyanhuang** [comment]
Updating PR based on comments

**Comment by tgohn** [comment]
:+1:

**Comment by zz85** [comment]
:+1:

**Pull request merged**